app/src/main/kotlin/com/wire/android/ui/home/conversations/details/GroupConversationDetailsViewModel.kt

```diff
@@ -91,13 +91,16 @@ class GroupConversationDetailsViewModel @Inject constructor(
         groupOptionsState = groupOptionsState.copy(loadingGuestOption = true, isGuestAllowed = enableGuestAndNonTeamMember)
         when (enableGuestAndNonTeamMember) {
             true -> updateGuestRemoteRequest(enableGuestAndNonTeamMember)
-            false -> showGuestConformationDialog()
+            false -> updateState(groupOptionsState.copy(changeGuestOptionConfirmationRequired = true))
         }
     }
 
     fun onServicesUpdate(enableServices: Boolean) {
         updateState(groupOptionsState.copy(loadingServicesOption = true, isServicesAllowed = enableServices))
-        updateServicesRemoteRequest(enableServices)
+        when(enableServices) {
```

Comment:
isn't simpler to use if :D

Reply:
I just like when it looks cleaner to me 
if there is another reason to use if over when, I will be happy to change it